# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/gl/GLContext.cpp
# Commit: 0067bb1fb8e4
# Full Hash: 0067bb1fb8e45078f4383c9e371dd1c7eafbffbe
# Author: Dzmitry Malyshau <dmalyshau@mozilla.com>
# Date: 2020-07-29 09:49:32
# Regressor Bug: 1652972
# File Overlap Count: 1
# Description:
#   Bug 1652972 - Mark EGL context as lost on Shutdown() r=aosmond
#   
#   This is meant to save us in cases where the message loop in GPU process
#   receives commands related to resources that point to the old EGL context
#   that was just shut down. Since the symbols are erased, we'd end up with
# ==============================================================================

diff -r bd33c30ba4b5 -r 0067bb1fb8e4 gfx/gl/GLContext.cpp
--- a/gfx/gl/GLContext.cpp	Tue Jul 28 22:26:21 2020 +0000
+++ b/gfx/gl/GLContext.cpp	Wed Jul 29 02:33:54 2020 +0000
@@ -2326,6 +2326,8 @@
 bool GLContext::MakeCurrent(bool aForce) const {
   if (MOZ_UNLIKELY(IsContextLost())) return false;
 
+  if (MOZ_UNLIKELY(!IsAliveImpl())) return false;
+
   if (MOZ_LIKELY(!aForce)) {
     bool isCurrent;
     if (mUseTLSIsCurrent) {